target/snitch_cluster/sw/apps/snax-xdma-memset/src/snax-xdma-memset.c

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 KU Leuven.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Yunhao Deng <yunhao.deng@kuleuven.be>
+
+#include "data.h"
+#include "snax-xdma-lib.h"
+#include "snrt.h"
+
+int main() {
+    // Set err value for checking
+    int err = 0;
+
+    // Obtain the start address of the TCDM memory
+    uint8_t *tcdm_baseaddress = (uint8_t *)snrt_l1_next();
+    uint8_t *tcdm_0 = tcdm_baseaddress;
+    uint8_t *tcdm_16 = tcdm_baseaddress + 0x4000 * sizeof(uint8_t);
+    uint8_t *tcdm_32 = tcdm_baseaddress + 0x8000 * sizeof(uint8_t);
+    uint8_t *tcdm_48 = tcdm_baseaddress + 0xc000 * sizeof(uint8_t);
+    uint8_t *tcdm_64 = tcdm_baseaddress + 0x10000 * sizeof(uint8_t);
+    uint8_t *tcdm_80 = tcdm_baseaddress + 0x14000 * sizeof(uint8_t);
+    uint8_t *tcdm_96 = tcdm_baseaddress + 0x18000 * sizeof(uint8_t);
+    uint8_t *tcdm_112 = tcdm_baseaddress + 0x1c000 * sizeof(uint8_t);
+
+    // Using xdma core only
+    if (snrt_cluster_core_idx() == snrt_cluster_compute_core_num() - 1) {
```

Comment:
Any particular reason why it has to be this long condition when you could just specify the actual core number where the xDMA is positioned? Or are you foreseeing that the xDMA will be placed in the 2nd to the last core always? However, that's not true right? Since we want the xDMA to be part of the accelerator cores too!

Reply:
xdma will finally replace idma, locating at the last. Now the development is not finished so idma cannot be removed.